controllers/utils/etcd.go

```diff
@@ -0,0 +1,38 @@
+package utils
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+
+	v1 "k8s.io/api/policy/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const etcdNamespace = "openshift-etcd"
+
+// +kubebuilder:rbac:groups="policy",resources=poddisruptionbudgets,verbs=get;list;watch
+
+// IsEtcdDisruptionAllowed checks if etcd disruption is allowed
+func IsEtcdDisruptionAllowed(ctx context.Context, cl client.Client, log logr.Logger) (bool, error) {
+	log = log.WithName("etcd-pdb-checker")
+	pdbList := &v1.PodDisruptionBudgetList{}
+	if err := cl.List(ctx, pdbList, &client.ListOptions{Namespace: etcdNamespace}); err != nil {
+		return false, err
+	}
+	if len(pdbList.Items) == 0 {
+		log.Info("No PDB found, can't check if etcd quorum will be violated! Refusing remediation!", "namespace", etcdNamespace)
+		return false, nil
+	}
+	if len(pdbList.Items) > 1 {
+		log.Info("More than one PDB found, can't check if etcd quorum will be violated! Refusing remediation!", "namespace", etcdNamespace)
```

Comment:
Why not check the pdb name or earlier just fetch PDB by name and not list all the PDBs in that namespace?
Its name is known as `etcd-guard-pdb` since OCP 4.11.

Reply:
> Its name is known as etcd-guard-pdb since OCP 4.11

Yes, but until when?
So we can bet on the name, or that exactly 1 exists. I went with the 2nd option, IMHO it's less likely to change (again)

Reply:
IIUC if someone (or another operator that runs in the background) decides to create in that namespace another PDB to `etcd-guard-pdb` then NHC won't allow the remediation of CP nodes.
I don't know what is less likely. I guess none of them but your call.

Reply:
> then NHC won't allow the remediation of CP nodes.

yes, and that will be catched by CI or QE, that's why I disallow in that case instead of allow. 🙂